Adamant/Modules/Chat/View/ChatViewController.swift

```diff
@@ -492,6 +525,16 @@ private extension ChatViewController {
         
         navigationItem.titleView?.addGestureRecognizer(tapGesture)
         navigationItem.titleView?.addGestureRecognizer(longPressGesture)
+        
+        view.addSubview(dateHeaderLabel)
+        dateHeaderLabel.backgroundColor = .adamant.chatSenderBackground
+        dateHeaderLabel.textInsets = .init(top: 4, left: 7, bottom: 4, right: 7)
+        dateHeaderLabel.layer.cornerRadius = 10
+        dateHeaderLabel.clipsToBounds = true
+        dateHeaderLabel.snp.makeConstraints { make in
+            make.top.equalTo(view.safeAreaLayoutGuide).offset(10)
+            make.centerX.equalToSuperview()
+        }
```

Comment:
Avoid defining views in huge view controller. There is too many code in this file. I suggest to move it to the other file